lib/ae_mdw_web/controllers/aex141_controller.ex

```diff
@@ -108,17 +108,26 @@ defmodule AeMdwWeb.Aex141Controller do
   end
 
   @spec owned_nfts(Conn.t(), map()) :: Conn.t() | {:error, ErrInput.t()}
-  def owned_nfts(%Conn{assigns: assigns} = conn, %{"account_id" => account_id}) do
+  def owned_nfts(%Conn{assigns: assigns} = conn, %{"account_id" => account_id} = params) do
     %{
       state: state,
       pagination: pagination,
       cursor: cursor
     } = assigns
 
     with {:ok, account_pk} <- Validate.id(account_id),
+         {:ok, contract_pk} <- validate_optional_pubkey(params, "contract"),
          {:ok, {prev_cursor, nfts, next_cursor}} <-
-           Aex141.fetch_owned_nfts(state, account_pk, cursor, pagination) do
+           Aex141.fetch_owned_nfts(state, account_pk, contract_pk, cursor, pagination) do
       WebUtil.paginate(conn, prev_cursor, nfts, next_cursor)
     end
   end
+
+  defp validate_optional_pubkey(params, param_name) do
+    case params |> Map.get(param_name) |> Validate.id() do
```

Comment:
Perhaps we can avoid having `Validate.id` handle any value type and use `Map.fetch(params, param_name)` here to tell if it's present or not